ledger/src/blockstore_processor.rs

```diff
@@ -1601,15 +1667,26 @@ pub mod tests {
             Ok(_)
         );
 
-        let (bank_forks, ..) = test_process_blockstore(
+        let (bank_forks, ..) = test_process_blockstore_with_custom_options(
             &genesis_config,
             &blockstore,
             ProcessOptions {
                 poh_verify: true,
                 ..ProcessOptions::default()
             },
+            blockstore_access_type.clone(),
         );
         assert_eq!(frozen_bank_slots(&bank_forks.read().unwrap()), vec![0]);
+
+        let dead_slots: Vec<Slot> = blockstore.dead_slots_iterator(0).unwrap().collect();
+        match blockstore_access_type {
+            AccessType::Secondary => {
+                assert_eq!(dead_slots.len(), 0);
```

Comment:
This is good!

Maybe consider adding a comment here describing if a DB is open as secondary then it will be immutable and thus we don't expect dead_slots to be updated.